### PR TITLE
Viewer: Update of the code, harmonization, cleanup, ...

### PR DESCRIPTION
**Pull request by lbocquet**
Viewer: Update of the code, harmonization, cleanup, ...

### CODE REVIEW

File: DSMPlugin/MSRC4Plugin/crypto.rc [hunk 1]
```diff
@@ -27,9 +52,12 @@ LANGUAGE LANG_ENGLISH, SUBLANG_ENGLISH_US
...
 
+// Defines STR_COMPANYNAME, STR_FILEVERSION, STR_PRODUCTVERSION, STR_COPYRIGHT,
+// STR_SPECIALBUILD, FILEVERSION, PRODUCTVERSION
+#include "../../common/versioninfo.h"
 VS_VERSION_INFO VERSIONINFO
- FILEVERSION 1,1,9,0
- PRODUCTVERSION 1,1,9,0
+ FILEVERSION INT_FILEVERSION
+ PRODUCTVERSION INT_PRODUCTVERSION
  FILEFLAGSMASK 0x3fL
```
Comment (RudiDeVos) [inline DSMPlugin/MSRC4Plugin/crypto.rc:61]: The plugin has it's own version.
1.1.9.0 is ok

**Comment by wqweto** [comment]
This PR is impossible to review. You had to split headers/comments update into an initial commit and then all code changes in one or more separate commits so these can be reviewed in isolation.

The way it is now no one can tell at a glance of all 35k modified lines which files have code changes and not some header/spelling being fixed or strings/messages being "harmonized".

File: DSMPlugin/MSRC4Plugin/version.rc [hunk 2]
```diff
@@ -1,14 +1,40 @@
...
 
+// Defines STR_COMPANYNAME, STR_FILEVERSION, STR_PRODUCTVERSION, STR_COPYRIGHT,
+// STR_SPECIALBUILD, FILEVERSION, PRODUCTVERSION
+#include "../../common/versioninfo.h"
 VS_VERSION_INFO VERSIONINFO
- FILEVERSION MAJOR_INT,MINOR_INT,BUILD_INT,SPECIAL_INT
- PRODUCTVERSION MAJOR_INT,MINOR_INT,BUILD_INT,SPECIAL_INT
+ FILEVERSION INT_FILEVERSION
+ PRODUCTVERSION INT_PRODUCTVERSION
  FILEFLAGSMASK 0x3fL
```
Comment (RudiDeVos) [inline DSMPlugin/MSRC4Plugin/version.rc:38]: The plugin has it own version

File: UltraVNC_installer_x64.iss
Comment (RudiDeVos) [inline UltraVNC_installer_x64.iss:1]: to many changes, need a seperated check

File: UltraVNC_installer_x86.iss
Comment (RudiDeVos) [inline UltraVNC_installer_x86.iss:1]: to many changes, need a seperated check

File: addon/ms-logon/MSLogonACL/.cvsignore
Comment (RudiDeVos) [inline addon/ms-logon/MSLogonACL/.cvsignore:1]: cvs is not longer used,we use git file can be removed

File: addon/ms-logon/MSLogonACL/MSLogonACLv11.vcxproj [hunk 3]
```diff
@@ -170,7 +170,7 @@
       <OutputFile>.\..\..\Release/MSLogonACL.bsc</OutputFile>
     </Bscmake>
     <PostBuildEvent>
-      <Command>C:\UltraVnc\codesigningv3\signtool.exe sign /f C:\UltraVnc\codesigningv3\uvnccert.pfx /p changeme /v /t http://timestamp.globalsign.com/scripts/timstamp.dll "$(OutDir)$(TargetName)$(TargetExt)"</Command>
+      <Command>C:\UltraVNC\codesigningv3\signtool.exe sign /f C:\UltraVNC\codesigningv3\uvnccert.pfx /p changeme /v /t http://timestamp.globalsign.com/scripts/timstamp.dll "$(OutDir)$(TargetName)$(TargetExt)"</Command>
```
Comment (RudiDeVos) [inline addon/ms-logon/MSLogonACL/MSLogonACLv11.vcxproj:173]: replave by
<Command></Command>
old signing not longer needed

File: addon/ms-logon/MSLogonACL/MSLogonACLv11.vcxproj [hunk 4]
```diff
@@ -211,7 +211,7 @@
...
```
Comment (RudiDeVos) [inline addon/ms-logon/MSLogonACL/MSLogonACLv11.vcxproj:214]: replace by
<Command></Command>
old signing not longer needed

File: addon/ms-logon/authSSP/.cvsignore
Comment (RudiDeVos) [inline addon/ms-logon/authSSP/.cvsignore:1]: can be removed

File: addon/ms-logon/authSSP/authSSPv11.vcxproj [hunk 5]
```diff
@@ -177,7 +177,7 @@
       <OutputFile>.\..\..\Release/authSSP.bsc</OutputFile>
     </Bscmake>
     <PostBuildEvent>
-      <Command>C:\UltraVnc\codesigningv3\signtool.exe sign /f C:\UltraVnc\codesigningv3\uvnccert.pfx /p changeme /v /t http://timestamp.globalsign.com/scripts/timstamp.dll "$(OutDir)$(TargetName)$(TargetExt)"</Command>
+      <Command>C:\UltraVNC\codesigningv3\signtool.exe sign /f C:\UltraVNC\codesigningv3\uvnccert.pfx /p changeme /v /t http://timestamp.globalsign.com/scripts/timstamp.dll "$(OutDir)$(TargetName)$(TargetExt)"</Command>
```
Comment (RudiDeVos) [inline addon/ms-logon/authSSP/authSSPv11.vcxproj:180]: postbuild can be removed

File: addon/ms-logon/authSSP/authSSPv11.vcxproj [hunk 6]
```diff
@@ -224,7 +224,7 @@
...
```
Comment (RudiDeVos) [inline addon/ms-logon/authSSP/authSSPv11.vcxproj:227]: postbuild can be removed

File: addon/ms-logon/authadm/authadminv11.vcxproj [hunk 7]
```diff
@@ -258,7 +258,7 @@
       <OutputFile>.\..\..\Release/authadmin.bsc</OutputFile>
     </Bscmake>
     <PostBuildEvent>
-      <Command>C:\UltraVnc\codesigningv3\signtool.exe sign /f C:\UltraVnc\codesigningv3\uvnccert.pfx /p changeme /v /t http://timestamp.globalsign.com/scripts/timstamp.dll "$(OutDir)$(TargetName)$(TargetExt)"</Command>
+      <Command>C:\UltraVNC\codesigningv3\signtool.exe sign /f C:\UltraVNC\codesigningv3\uvnccert.pfx /p changeme /v /t http://timestamp.globalsign.com/scripts/timstamp.dll "$(OutDir)$(TargetName)$(TargetExt)"</Command>
```
Comment (RudiDeVos) [inline addon/ms-logon/authadm/authadminv11.vcxproj:261]: All v11 projects will be removed, no changes needed

File: addon/versioninfo.h
Comment (RudiDeVos) [inline addon/versioninfo.h:1]: There is no common version for all exe and dll's.
To avoid unneeded updates we keep the current version of the files.
When an update is made, the new versioning is used
year-month-day-buildnummer

File: common/versioninfo.h
Comment (RudiDeVos) [inline common/versioninfo.h:1]: who is using this file, with the new versioning is this file still needed, as common there is only the company name.
Build version will not be the same for server and viewer it's an incremental nummer augmented on each change

File: repeater/resources.rc
Comment (RudiDeVos) [inline repeater/resources.rc:1]: The repeater has his own numbering, doesn't have to be in sync with the server of viewer

File: repeater/repeater.vcproj
Comment (RudiDeVos) [inline repeater/repeater.vcproj:1]: Project broken, files not found update sln and vcxproj not correct

File: uvnc_settings2/uvnc_settings.sln
Comment (RudiDeVos) [inline uvnc_settings2/uvnc_settings.sln:1]: Project doesn't load error  in sln or vcxproj

File: vncviewer/ClientConnection.cpp [hunk 8]
```diff
@@ -77,8 +78,8 @@ extern "C" {
 #else
 #define MAX_ENCODINGS (LASTENCODING+52)
 #endif
-#define VWR_WND_CLASS_NAME _T("VNCviewer")
-#define VWR_WND_CLASS_NAME_VIEWER _T("VNCviewerwindow")
+#define VWR_WND_CLASS_NAME _T("ultravnc-viewer")
+#define VWR_WND_CLASS_NAME_VIEWER _T("ultravnc-viewer-window")
```
Comment (RudiDeVos) [inline vncviewer/ClientConnection.cpp:82]: Other software connect to the viewer like this
vb. sample token from the net
WinHandle = FindWindow("vncviewer", vbNullString)

If we change the window name, they all need to change the name.
Even when the name is bad, we are not going to touch it.

File: vncviewer/res/vncviewer.rc
Comment (RudiDeVos) [inline vncviewer/res/vncviewer.rc:1]: has his own versioning yy-mm-dd buildnummer

File: winvnc/UltraVNC+/UltraVNC+_VS2015_x64.vdproj
Comment (RudiDeVos) [inline winvnc/UltraVNC+/UltraVNC+_VS2015_x64.vdproj:1]: msi wil  is not generated
1) viewer postbuild copy vncviewer.exe and vncviewer64.exe in Files, msi look in the wrong folder
2) winvnc.exe looks for xxx64.dll while files now call the same as for 32 xxx.dll
All need to be sync or it will not work

File: winvnc/UltraVNC+/UltraVNC+_VS2013_x86.vdproj
Comment (RudiDeVos) [inline winvnc/UltraVNC+/UltraVNC+_VS2013_x86.vdproj:1]: msi wil  is not generated
1) viewer postbuild copy vncviewer.exe  in Files, msi look in the wrong folder Files/x86

File: winvnc/UltraVNC+/UltraVNC+_VS2015_x86.vdproj
Comment (RudiDeVos) [inline winvnc/UltraVNC+/UltraVNC+_VS2015_x86.vdproj:1]: msi wil is not generated

viewer postbuild copy vncviewer.exe in Files, msi look in the wrong folder Files/x86

File: winvnc/UltraVNC+/UltraVNC+_VS2010_x64.vdproj
Comment (RudiDeVos) [inline winvnc/UltraVNC+/UltraVNC+_VS2010_x64.vdproj:1]: To be removed

File: winvnc/UltraVNC+/UltraVNC+_VS2013_x64.vdproj
Comment (RudiDeVos) [inline winvnc/UltraVNC+/UltraVNC+_VS2013_x64.vdproj:1]: remove

File: winvnc/UltraVNC+/UltraVNC+_VS2010_x86.vdproj
Comment (RudiDeVos) [inline winvnc/UltraVNC+/UltraVNC+_VS2010_x86.vdproj:1]: Remove 2010 2013 are not needed

File: winvnc/winvnc/vncdesktop.cpp [hunk 9]
```diff
@@ -62,7 +64,7 @@ int getinfo(char mytext[1024]);
 const UINT RFB_SCREEN_UPDATE = RegisterWindowMessage("WinVNC.Update.DrawRect");
 const UINT RFB_COPYRECT_UPDATE = RegisterWindowMessage("WinVNC.Update.CopyRect");
 const UINT RFB_MOUSE_UPDATE = RegisterWindowMessage("WinVNC.Update.Mouse");
-const char szDesktopSink[] = "WinVNC desktop sink";
+const char szDesktopSink[] = "UltraVNC Server desktop sink";
```
Comment (RudiDeVos) [inline winvnc/winvnc/vncdesktop.cpp:68]: Don't change
This name is used by other softwares.
You changed it to the same in vnchook and winvnc ..correct,
But changed it require a version check between both, older hooks will fail with new server and new hook wil fail on old server.  The risk for possible mistakes it to high

File: winvnc/vnchooks/VNCHooks.cpp [hunk 10]
```diff
@@ -463,18 +465,18 @@ inline void SendUpdateRect(SHORT x, SHORT y, SHORT x2, SHORT y2)
...
-	// Send the update to WinVNC
+	// Send the update to UltraVNC Server
 	PostThreadMessage(
 		 g_shdata->vnc_thread_id,
 		 g_shdata->UpdateRectMessage,
 		vwParam,
 		vlParam
 		);
-	HWND hWndRemote=FindWindow("WinVNC desktop sink", "WinVNC");
+	HWND hWndRemote=FindWindow("UltraVNC Server desktop sink", "WinVNC");
```
Comment (RudiDeVos) [inline winvnc/vnchooks/VNCHooks.cpp:475]: Don't change
This name is used by other softwares.
You changed it to the same in vnchook and winvnc ..correct,
But changed it require a version check between both, older hooks will fail with new server and new hook wil fail on old server. The risk for possible mistakes it to high

File: winvnc/vnchooks/vnchooks.rc
Comment (RudiDeVos) [inline winvnc/vnchooks/vnchooks.rc:1]: Company and productname is changed
File wil be considered as new by all virus software, until critical mass is passed it will be flagged as potential risk
( hooking is often used by hacker)
vnchook has it own versioning system
I need to look further, possible impact on versioning system

File: winvnc/winvnc/DeskdupEngine.cpp [hunk 11]
```diff
@@ -28,33 +53,20 @@ DeskDupEngine::DeskDupEngine()
 	HideCursorW8 = NULL;
 	osVer = osVersion();
 	if (osVer == OSWIN10) {
-#ifdef _X64
 		char szCurrentDir[MAX_PATH];
```
Comment (RudiDeVos) [inline winvnc/winvnc/DeskdupEngine.cpp:56]: As i told many times we use different names for the same dll when there is a risk that a x64 app can load a 32-bit dll.
We had to many bug requests in the past for this easy to avoid issue.

File: winvnc/winvnc/VirtualDisplay.cpp [hunk 12]
```diff
@@ -445,11 +470,7 @@ bool VirtualDisplay::InstallDriver(bool fromCommandline)
 			if (p == NULL)
 				return 0;
 			*p = '\0';
-#ifdef _X64
-			strcat_s(szdriverPath, "\\UVncVirtualDisplay64\\UVncVirtualDisplay.inf");
-#else
 			strcat_s(szdriverPath, "\\UVncVirtualDisplay\\UVncVirtualDisplay.inf");
```
Comment (RudiDeVos) [inline winvnc/winvnc/VirtualDisplay.cpp:473]: Perhaps we need to call the x64 files UVncVirtualDisplay64.inf and place then in a folder UVncVirtualDisplay
But not the same location and name for 64 and 32-bit

File: winvnc/winvnc/initipp.cpp
Comment (RudiDeVos) [inline winvnc/winvnc/initipp.cpp:1]: file can be removed, ipp is not longer used  
and references in other files
#ifdef IPP

File: winvnc/winvnc/winvnc.rc [hunk 13]
```diff
@@ -487,9 +512,12 @@ IDC_CURSOR3             CURSOR                  "res/cursor3.cur"
 // Version
 //
 
+// Defines STR_COMPANYNAME, STR_FILEVERSION, STR_PRODUCTVERSION, STR_COPYRIGHT,
+// STR_SPECIALBUILD, FILEVERSION, PRODUCTVERSION
+#include "../../common/versioninfo.h"
 VS_VERSION_INFO VERSIONINFO
```
Comment (RudiDeVos) [inline winvnc/winvnc/winvnc.rc:518]: file version  is auto generated yy mm dd

File: winvnc/winvnc/winvnc.rc [hunk 14]
```diff
@@ -504,17 +532,23 @@ BEGIN
...
+            VALUE "LegalCopyright", STR_COPYRIGHT
             VALUE "OriginalFilename", "WinVNC.exe"
-            VALUE "PrivateBuild", "1.4.4.0"
-            VALUE "ProductName", "UltraVNC"
-            VALUE "ProductVersion", "2024.02.17.3"
+            VALUE "PrivateBuild", ""
+            VALUE "ProductName", "UltraVNC Server"
+            VALUE "ProductVersion", STR_PRODUCTVERSION
+#if isRC == 1
+			VALUE "SpecialBuild", STR_SPECIALBUILD
```
Comment (RudiDeVos) [inline winvnc/winvnc/winvnc.rc:550]: Don't know the impact of changing productname and comapny name.
This is another software with the same filename.
